derive/src/to_delegate.rs

```diff
@@ -232,28 +250,38 @@ impl RpcMethod {
 			.take(param_types.len())
 			.map(|name| ident(name))
 			.collect());
-		let param_types = &param_types;
-		let parse_params = {
-			// last arguments that are `Option`-s are optional 'trailing' arguments
-			let trailing_args_num = param_types.iter().rev().take_while(|t| is_option_type(t)).count();
-
-			if trailing_args_num != 0 {
-				self.params_with_trailing(trailing_args_num, param_types, tuple_fields)
-			} else if param_types.is_empty() {
-				quote! { let params = params.expect_no_params(); }
-			} else if self.attr.params_style == Some(ParamStyle::Raw) {
-				quote! { let params: _jsonrpc_core::Result<_> = Ok((params,)); }
-			} else if self.attr.params_style == Some(ParamStyle::Positional) {
-				quote! { let params = params.parse::<(#(#param_types, )*)>(); }
-			} else {
-				unimplemented!("Server side named parameters are not implemented");
+		let parse_params = if param_types.is_empty() {
+			quote! { let params = params.expect_no_params(); }
+		} else {
+			match self.attr.params_style.as_ref().unwrap() {
+				ParamStyle::Raw => quote! { let params: _jsonrpc_core::Result<_> = Ok((params,)); },
+				ParamStyle::Positional => {
+					// last arguments that are `Option`-s are optional 'trailing' arguments
+					let trailing_args_num = param_types.iter().rev().take_while(|t| is_option_type(t)).count();
+					if trailing_args_num != 0 {
+						self.params_with_trailing(trailing_args_num, &param_types, tuple_fields)
+					} else {
+						quote! { let params = params.parse::<(#(#param_types, )*)>(); }
+					}
+				}
+				ParamStyle::Named => quote! {
+					#[derive(serde::Deserialize)]
+					#[allow(non_camel_case_types)]
+					struct __Params {
+						#(
+							#fn_args,
+						)*
+					}
+					let params = params.parse::<__Params>()
+						.map(|__Params { #(#arg_names, )* }| (#(#arg_names, )*));
+				},
 			}
 		};
 
 		let method_ident = self.ident();
 		let result = &self.trait_item.sig.output;
-		let extra_closure_args: &Vec<_> = &special_args.iter().cloned().map(|arg| arg.0).collect();
-		let extra_method_types: &Vec<_> = &special_args.iter().cloned().map(|arg| arg.1).collect();
+		let extra_closure_args: Vec<_> = special_args.iter().cloned().map(|arg| *arg.pat).collect();
+		let extra_method_types: Vec<_> = special_args.iter().cloned().map(|arg| *arg.ty).collect();
```

Comment:
same comments about `cloned()`, maybe disregard the previous suggestion to add generic parameter to `collect` to keep consistency.